src/main/kotlin/org/wycliffeassociates/otter/common/recorder/PCMCompressor.kt

```diff
@@ -0,0 +1,43 @@
+package org.wycliffeassociates.otter.common.app
+
+import org.wycliffeassociates.otter.common.collections.FloatRingBuffer
+
+class PCMCompressor(private val ringBuffer: FloatRingBuffer) {
+
+    //size is how small the waveform is being shrunk for visualization
+    //arbitrary for now, could be based on number of seconds to show and resolution
+    val accumulator = FloatArray(480)
+    var index = 0
+
+    fun add(data: FloatArray) {
+        for(sample in data) {
+            if(index >= accumulator.size){
+                sendDataToRingBuffer()
+                index = 0
+            }
+            accumulator[index] = sample
+            index++
+        }
+    }
+
+    fun add(data: Float) {
+        if(index >= accumulator.size){
+            sendDataToRingBuffer()
+            index = 0
+        }
+        accumulator[index] = data
+        index++
+    }
+
+    fun sendDataToRingBuffer() {
+        var min = Float.MAX_VALUE
+        var max = Float.MIN_VALUE
+
+        for(sample in accumulator) {
+            max = if(max < sample) sample else max
```

Comment:
Maybe kotlin has a function to find max and min?

Reply:
fixed, thanks!